src/rgw/rgw_op.cc

```diff
@@ -2213,6 +2213,21 @@ void RGWGetObj::execute()
   gc_invalidate_time = ceph_clock_now();
   gc_invalidate_time += (s->cct->_conf->rgw_gc_obj_min_wait / 2);
 
+  if (s->info.args.exists("select-type"))
```

Comment:
maybe better to override the `execute()` call in the derived class, instead of checking that the argument exist here

Reply:
it means not to reuse the execute.

Reply:
yes. see: https://github.com/ceph/ceph/pull/33228#discussion_r378708674